package/ota_daemon/src/ota_daemon.c

```diff
@@ -102,7 +99,7 @@ static int parse_options(int argc, char *argv[])
     {"debug",     no_argument,        0, OPT_ID_DEBUG},
     {"ota",       no_argument,        0, OPT_ID_MODE_OTA},
     {"settings",  no_argument,        0, OPT_ID_MODE_SETTINGS},
-    {"url",       optional_argument,  0, OPT_ID_URL},
+    {"url",       required_argument,  0, OPT_ID_URL},
```

Comment:
I presume that you are thinking that `required_argument` would mean that `--url` switch would be mandatory for the OTA daemon? This is not the case, this only means that in case of `--url` is given, it is mandatory to give an argument after it. If `--url` is not present, it's fine and default URL is used.

Reply:
Oops, yeah, duh! 🙂